cdap-api/src/main/java/co/cask/cdap/api/annotation/Capabilities.java

```diff
@@ -16,12 +16,25 @@
 package co.cask.cdap.api.annotation;
 
 /**
- * Defines the different capability options for {@link Plugin} to specify.
+ * <p>Defines the different capability options for {@link Capability} to specify.
+ * These are system defined capabilities.</p>
+ *
+ * <p>If {@link Capability} does not specifically define any of the capability from {@link Capabilities} then
```

Comment:
Is this true? Meaning if a plugin is annotated with `@Capability("xyz")` then it will implicitly have `native` and `remote`?

Reply:
Or should the statement be `If a {@link Plugin} does not define any {@link Capability}, then the {@link Capability#default} system capability will be assumed`?

Reply:
if a plugin only annotated with 'xyz' does not get any system capabilities, then it means it can't actually be used anywhere, which seems weird. 

maybe we should differentiate between system and custom capabilities? If no system capability is given, it gets all system capabilities.

Reply:
If the set of system capabilities is keep expanding, it might be hard for plugin to catch up with.

Reply:
I think the difficulty comes from the fact that we're trying to use the same annotation for two separate use cases. For system things, we really want the plugin to specify what it requires, not what it's capable of. When we run, everything a plugin requires must be supported by the provisioner. 

And for custom things, it's the opposite. We want to specify what capabilities it has, so that an admin can configure CDAP to require a set of capabilities.

What if we separate these things? Provisioners are responsible for specifying what capabilities they have. Native would have {transactional}, dataproc would have {gcp}, emr would have {aws}. Transactional plugins would be annotated as transactions. Plugins that work anywhere would not need any annotations.

For the pii case, custom plugins would annotate that they have a 'pii' capability.

At a CDAP level, you can configure a whitelist of plugin capabilities. If you want only pii plugins in CDAP, configure 'plugin.capability.whitelist' to 'pii'.
You can also configure a blacklist of plugin requirements. If you don't want transactional plugins to appear, configure 'plugin.requirement.blacklist' to 'transactional'.

Reply:
Yes, if a plugin does not specify any capability from System `Capabilities` it will be considered to be capable of all system defined capabilities. This applies even if they only specify custom capabilities.

Reply:
```
If the set of system capabilities is keep expanding, it might be hard for plugin to catch up with.
```

In this case the plugin will specifically specify what they are capable once (when they are no more capable of all system defined capabilities). Once it is specified as long as now new capability is added to the plugin the specification does not need to be changed. If new capability is added to the plugin then we can expect the plugin developer to a update the specification too.

Reply:
We had in person discussion and decided that we will separate out the annotation for system capabilities and user capabilities. I will update the design with this new decision and update the code here.

Reply:
https://wiki.cask.co/display/CE/Improving+Plugins+User+Experience+Across+Different+Modes?focusedCommentId=8716406#comment-8716406

Reply:
@chtyim @albertshau  Update on design: https://wiki.cask.co/display/CE/Improving+Plugins+User+Experience+Across+Different+Modes?focusedCommentId=8716406#comment-8716406

Please have a look. I will update the code with this shortly.